addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationJob.java

```diff
@@ -307,47 +310,127 @@ protected void addTests(Object testsObj, AutomationProgress progress) {
         for (Object testObj : (ArrayList<?>) testsObj) {
             if (!(testObj instanceof LinkedHashMap<?, ?>)) {
                 progress.error(Constant.messages.getString("automation.error.job.data", testObj));
                 continue;
             }
             LinkedHashMap<?, ?> testData = (LinkedHashMap<?, ?>) testObj;
             Object testType = testData.get("type");
-            if ("stats".equals(testType)) {
-                String statistic = safeCast(testData.get("statistic"), String.class);
-                String operator = safeCast(testData.get("operator"), String.class);
-                Number number = safeCast(testData.get("value"), Number.class);
-                String onFail = safeCast(testData.get("onFail"), String.class);
-                if (statistic == null || operator == null || number == null || onFail == null) {
+
+            switch ((String) testType) {
+                case "stats":
+                    addStatisticTest(testData, progress);
+                    break;
+                case "alert":
+                    if (getType().equals(ActiveScanJob.JOB_NAME)
+                            || getType().equals(PassiveScanWaitJob.JOB_NAME)) {
+                        addAlertTest(testData, progress);
+                    } else {
+                        progress.error(
+                                Constant.messages.getString(
+                                        "automation.tests.alert.invalidJobType", getType()));
+                    }
+                    break;
+                default:
                     progress.warn(
-                            Constant.messages.getString(
-                                    "automation.tests.stats.missingOrInvalidProperties",
-                                    getType()));
-                    continue;
-                }
-                long value = number.longValue();
-                String name = safeCast(testData.get("name"), String.class);
-                if (name == null || name.isEmpty()) {
-                    name = statistic + ' ' + operator + ' ' + value;
-                }
-                progress.info(
-                        Constant.messages.getString("automation.tests.stats.add", getType(), name));
-                try {
-                    addTest(
-                            new AutomationStatisticTest(
-                                    statistic, name, operator, value, onFail, getType()));
-                } catch (IllegalArgumentException e) {
-                    progress.warn(e.getMessage());
-                    return;
-                }
-            } else {
-                progress.warn(
-                        Constant.messages.getString("automation.tests.invalidType", testType));
+                            Constant.messages.getString("automation.tests.invalidType", testType));
             }
         }
     }
 
+    private void addStatisticTest(LinkedHashMap<?, ?> testData, AutomationProgress progress) {
```

Comment:
I dont think AutomationJob should know anything about the test classes other the absolute minimum.
The parsing should be in the relevant test classes, and ideally the classes should be registered with a key that the job / infra can use to link to the relevant class. This will then allow add-ons to add test types without having to change the authentication classes.